### ISSUE TITLE
The exported image is grayscale.

### ISSUE THREAD

**Issue by tama34** [ilastik/ilastik#2448]
## Describe the bug
I tried to export the segmentation map created in Pixel Classification to Object Classification. 
Normally, the segmentation area should be colored, but it turned out to be a grayscale image. 
How can I get them colored?

## Expected behavior
When exporting the segmentation map created in Pixel Classification to Object Classification, I thought the image would be colored.

## Screenshots
If you need screenshots, I would be happy to send them to you privately.

## Desktop (please complete the following information):
 - ilastik version: 1.3.3
 - OS: Windows 10

**Comment by k-dominik** [comment]
Hello @tama34,

this is a pretty common issue a lot user run into. See point 4 in our faq:

> 4) My exported results are all black!
>
> ilastik exports probability maps or segmentations. The former have a range from 0 to 1 and by default they are of type float. This is convenient for automatic post-processing, but not for visual inspection. To export as a viewable image, change the output type to unsigned int, 8 bit and renormalize the range from (0, 1) to (0, 255). More details on these operations can be found here. If you are exporting a segmentation, ilastik saves it as a labeled image, so every pixel has the value of the most probable label found during classification. If you had, say, 3 labels, your image is composed of values 1, 2 and 3, which looks black if viewed raw. An image like this should be displayed with a lookup table. For example, glasbey LUT in Fiji usually works well, but you can also define your own if you have specific class colors in mind.

Side note on using the combined pixel and object classification workflow (see also [Point 9 in the faq](https://www.ilastik.org/documentation/basics/common_problems#9-pixelobject-classification-workflow-is-slowconsumes-too-much-ram): For anything serious, we recommend using the two workflows separately, using the output from pixel classification in an object classification project. The combined workflow consumes a lot of RAM. Furthermore, for this worklfow I'd highly recommend using the latest ilastik beta version.

**Comment by tama34** [comment]
The question was viewed and implemented as described. 
However, the prediction map displayed on object classification became grayscale.

**Comment by k-dominik** [comment]
Hey @tama34,

we don't "include" any color information in the exported output. So interpreting them as grayscale is solid default.
If you open them in a fiji, e.g. this is the expected behavior. You can apply colors via a colormap.

**Comment by tama34** [comment]
Thank you for your reply.
I watched a tutorial video uploaded on youtube. 
When importing a prediction map from pixel classification onto object classification, the map was a segmented colored image. 
Why would it be colored?

**Comment by k-dominik** [comment]
there we just assume it came from ilasitk and color it. There is no color information in the image.

**Comment by tama34** [comment]
https://www.youtube.com/watch?v=SQeRGvHeT3o

You're referring to the image at 0:25 in this video, right?
I have probably imported a colored image similar to this one in object classification.

**Comment by k-dominik** [comment]
> https://www.youtube.com/watch?v=SQeRGvHeT3o
> 
> You're referring to the image at 0:25 in this video, right?
> I have probably imported a colored image similar to this one in object classification.

yes. exactly. ilastik finds a 2-channel image, and applies default for this (RGB, so red green blue - only the first two channels are used, that's why it is red green).

**Comment by tama34** [comment]
I currently classify 4 different classes.
Is it still possible for me to import coloured images into object classification?

**Comment by k-dominik** [comment]
Hey @tama34,

you can import multi-channel images into object classification, both as raw data, and as prediction maps.

**Comment by tama34** [comment]
When the four classifications were made, the raw data were naturally grayscale, but the prediction maps were also exported to object classification in grayscale.

**Comment by k-dominik** [comment]
Hey @tama34,

this again is purely a display issue. In order to render color (when loading the file - no further information available), ilastik _tries_ to do something sensible. If you have an image with
* a single channel: -> it will be rendered as grayscale
* 2-3 channels: -> rgb space is used. first channel red, second green, third blue
* 4+ channels: -> grayscale with selector for the channel in the layers stack (bottom left).

you can influence to some extend how data is shown (see [our docs](https://www.ilastik.org/documentation/basics/dataselection#properties)). E.g. you could change 2-3 channels from RGB-Colors to grayscale with channel selector...

in thresholding (there one "knows" how the data should be displayed), ilastik will interpret the data already as a probability map and show it as an alpha blended image with default label colors.

I can see how this different display of data can be irritating. But there are reasons we don't want "upstream" applets to know too much about  downstream stuff in order to keep stuff modular.

Bottom line is: How data is displayed in ilastik does _not change_ the data.

**Comment by tama34** [comment]
I have actually exported to object classification with four different classifications. 
It works as you said.

I understand it to a certain extent. 
Thank you very much for your detailed answer.

**Comment by k-dominik** [comment]
okay cool. I'll close this issue for now.

By the way, do you know the bio-image analysis community forum https://forum.image.sc ? We are there, too and it is a better fit for ilastik usage questions.
